tests/test_merge_cdxj.py

```diff
@@ -0,0 +1,147 @@
+    original_obj = {'mime': 'text/html'}
+    meta_obj = {'Identified-Payload-Type': {'fido': 'image/gif', 'python-magic': 'image/gif'},
+                'Preservation-Identifier': 'fmt/4'}
+    actual = merge_cdxj.get_sidecar_fields(original_obj, meta_obj)
+    assert actual == {'mime': 'text/html', 'mime-detected': 'image/gif'}
+    m_lang.assert_not_called()
+
+
+@patch('merge_cdxj.get_alpha3_language_codes')
+def test_merge_meta_fields(m_lang):
+    m_lang.return_value = 'eng'
+    lang_array = [{'name': 'ENGLISH', 'code': 'en', 'text-covered': 99, 'score': 971.0}]
+    cdxj = io.StringIO('com,example) 20091111212121 {"url": "http://www.example.com", \
+                                                     "mime": "text/html"}\n', newline='\n')
+    original, edited, non_edited = merge_cdxj.merge_meta_fields(META_DICT, cdxj)
+    assert original == MERGED_LIST
+    assert not non_edited
+    assert edited == 1
+    m_lang.assert_called_once_with(lang_array)
+
+
+@patch('merge_cdxj.get_sidecar_fields')
+def test_merge_meta_fields_non_edited(m_fields):
+    m_fields.return_value = {'url': 'http://www.example.com',
+                             'mime': 'text/html',
+                             'mime-detected': 'text/html',
+                             'charset': 'ascii',
+                             'languages': 'eng',
+                             'soft-404-detected': 0.08195022044249829}
+    cdxj = io.StringIO('com,example) 20091111212121 {"url": "http://www.example.com", \
+                                                     "mime": "text/html"}\n'
+                       'com,abc) 20091111212131 {"url": "http://www.abc.com", '
+                       '"mime": "text/xml"}\n', newline='\n')
+    merged_list = ['com,example) 20091111212121 {"url": "http://www.example.com", '
+                   '"mime": "text/html", "mime-detected": "text/html", '
+                   '"charset": "ascii", "languages": "eng", '
+                   '"soft-404-detected": 0.08195022044249829}\n',
+                   'com,abc) 20091111212131 {"url": "http://www.abc.com", "mime": "text/xml"}\n']
+    original, edited, non_edited = merge_cdxj.merge_meta_fields(META_DICT, cdxj)
+    assert original == merged_list
+    assert non_edited == 1
+    assert edited == 1
+    m_fields.assert_called_once()
+
+
+def test_create_dict_from_meta():
+    actual_dict = merge_cdxj.create_dict_from_meta(META_FILE)
+    assert actual_dict == META_DICT
+
+
+def test_create_cdxj_path(tmpdir):
+    cdxj_path = merge_cdxj.create_cdxj_path('home/file.cdxj', str(tmpdir))
+    expected_path = os.path.join(tmpdir / 'file_merged.cdxj')
+    assert cdxj_path == expected_path
+
+
+class Test_Merge_Cdxj:
+    @patch('merge_cdxj.create_dict_from_meta')
+    @patch('merge_cdxj.merge_meta_fields')
+    @patch('builtins.open', spec=open)
+    def test_merge_cdxj2(self, mock_open, m_merge_meta, m_create_dict, caplog, tmpdir):
+        caplog.set_level(INFO)
+        merged_file = io.StringIO()
```

Comment:
We set this io object up, which is where it looks like the results of the merging will go, but then we never check the contents of it. Should we be doing some assertions on this later on? Otherwise it doesn't really seem like we've done anything to check that this function generates what we expect it to, just that we've called some expected functions within it and generated some logs.